pkg/utils/machines.go

```diff
@@ -25,3 +32,19 @@ func GetMasterMachines(kclient client.Client) (*machineapi.MachineList, error) {
 	}
 	return machineList, nil
 }
+
+func GetControlPlaneMachineSet(kclient client.Client) (*machinev1.ControlPlaneMachineSet, error) {
+	cpms := &machinev1.ControlPlaneMachineSet{}
+	key := client.ObjectKey{
+		Namespace: machineApiNamespace,
+		Name:      cpmsName,
+	}
+	err := kclient.Get(context.TODO(), key, cpms)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil, nil // Nothing to do
+		}
```

Comment:
This behavior kinda depends on the use-case for this function, I would prefer to see why this function is needed first before we implement the function.

Reply:
It's just a helper to prep for https://issues.redhat.com/browse/OSD-14568. Would you like me to delete this function until that ticket is fully implemented? (at which point I'd rather just close this PR entirely)

Reply:
+1 to remove this function until we can understand (and review) it with the full logic flow

Reply:
+1 to remove this function as well. It should be merged in relevant PR should it be necessary

Reply:
This work was initially going to be for the whole ticket, but after running into so many dependency issues I decided to split out this PR separately. Michael's PR #310 supersedes that effort, so there's no real reason to keep this PR around. Going to close it.